File: NeuroamWPF/Neuroam/Neuroam/source/QueryHandler.cs
using System.Windows.Controls;
using NeuroamCore;

namespace Neuroam
{
    public class QueryHandler
    {
        // User search What's on your mind?
        string CurrentQuery { get; set; }
        ListBox m_ResultsListBox;
        QueryDictionary m_QueryDictionary = new QueryDictionary();

        public QueryHandler(ListBox resultListBox)
        {
            m_ResultsListBox = resultListBox;
        }

        public void OnSearchChanged(string inFlightQuery)
        {
            if (!inFlightQuery.Equals(CurrentQuery))
            {
                CurrentQuery = inFlightQuery;
                m_QueryDictionary.Query(CurrentQuery);
                m_ResultsListBox.ItemsSource = m_QueryDictionary.Find(CurrentQuery);
            }
        }
        public void OnFinalizeQuery()
        {
            m_QueryDictionary.Query(CurrentQuery, true /*Add this query to the dictionary*/);
            m_ResultsListBox.ItemsSource = m_QueryDictionary.Find(CurrentQuery);
        }

        public void OnClose()
        {
            m_QueryDictionary.OnClose();
        }
    }
}

File: Neuroam/source/QueryHandler.cpp
#include <PrecompiledHeader.h>

#include <QueryHandler.h>

namespace Neuroam {
QueryHandler::QueryHandler(bool inMemory)
{
    m_QueryDictionary = new QueryDictionary(inMemory);
}

QueryHandler::~QueryHandler()
{
    delete m_QueryDictionary;
}

void QueryHandler::OnSearchChanged(StringView inFlightQuery)
{
    if (!m_CurrentQuery.Equals(inFlightQuery))
    {
        // Tokenize and search again
        m_CurrentQuery = inFlightQuery;
    }
}

void QueryHandler::OnFinalizeQuery()
{
    m_QueryDictionary->Add(m_CurrentQuery.GetRaw());
    Strings results;
    if(m_QueryDictionary->Find(m_CurrentQuery.GetRaw(), results))
    {
        for (const auto& result : results) {
            Logger("%s", result.ToString().data());
        }
    }
    
}

void QueryHandler::OnClose()
{
    m_QueryDictionary->OnClose();
}
}